src/hx/Thread.cpp

```diff
@@ -3,7 +3,7 @@
 #include <hx/Thread.h>
 #include <time.h>
 
-#ifdef HX_WINRT
+#if defined(HX_WINRT) && (_WIN32_WINNT == _WIN32_WINNT_WIN8)

```

Comment:
Should this be checking for `__cplusplus_winrt` instread of `_WIN32_WINNT_WIN8`? Maybe I misunderstand, but these are WinRT APIs, so building with the WinRT runtime would be necessary.

Also the same for similar changes in the rest of the PR.